templates/common/infra/bicep/core/database/postgresql/flexibleserver.bicep

```diff
@@ -0,0 +1,63 @@
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+
+param sku object
+param storage object
+param administratorLogin string
+@secure()
+param administratorLoginPassword string
+param databaseNames array = []
+param allowAzureIPsFirewall bool = false
+param allowAllIPsFirewall bool = false
+param allowedSingleIPs array = []
+
+// PostgreSQL version
+param version string
+
+resource postgresServer 'Microsoft.DBforPostgreSQL/flexibleServers@2022-12-01' = {
```

Comment:
@weikanglim 
It totally works without '-preview', except that annoyingly the types have not been updated yet - apparently the team needs to publish their swagger spec still.

That means you'll see warnings in VS Code:

<img width="792" alt="Screenshot 2023-01-30 at 3 18 07 PM" src="https://user-images.githubusercontent.com/297042/215618642-0eb35aa1-f6b2-482a-81a7-7a1888771898.png">

The lack of types is reported here:
https://github.com/Azure/bicep/issues/784

I think it's still fine to go with the "-preview" version, since the types should get updated soon enough.

Reply:
Let's go with "-preview", but leave a comment that `// latest official version 2022-12-01 does not have bicep types available`

Reply:
Ah, shoot, even preview of 2022-12-01 doesn't have types. The version I had before was from January and did have types. Does that change your preference?

Reply:
January 2023? As long as it's recent enough, I'm okay with it.

Reply:
The most recent one that has types is 2022-03-08-preview. You can see in the changelog what has changed since then:
https://learn.microsoft.com/en-us/azure/templates/microsoft.dbforpostgresql/change-log/flexibleservers

Neither of those properties are currently used by this module.

Reply:
OK. Let's just keep the one we have since it works. The type definitions are way out of our control, and it looks like the service team just needs to do better.